hack/docker.sh

```diff
@@ -27,6 +36,13 @@ docker-login() {
     local image=$1
     local registry=${image%%"/"*}
 
+    # Since this check doesn't work very well in Jenkins, and Jenkins sets
+    # environment variable 'CI' only perform this check when not in Jenkins.
+    if [[ -z "${CI}" && -f ~/.docker/config.json && $(grep -q "${registry}" ~/.docker/config.json) ]]; then
```

Comment:
```suggestion
    if [[ -z "${CI}" && -f ~/.docker/config.json ]] && grep -q "${registry}" ~/.docker/config.json; then
```

Reply:
Not sure it's still relevant to keep this check with [a4c46c2](https://github.com/elastic/cloud-on-k8s/pull/5665/commits/a4c46c265028ae8fbb19a2ba111bee2a57dc3315) but if we want it, this should be the fix to make shellcheck happy.